docs: expand on recommendations for CPU resource reservation (#25964)
hashicorp/nomad: website/content/docs/job-specification/resources.mdx

### Cores

This example specifies that the task requires 2 reserved cores. With this block,

> Nomad will find a client with enough spare capacity to reserve 2 cores

Reviewer: ```suggestion
Nomad finds a client with enough spare capacity to reserve 2 cores
```

> exclusively for the task. Unlike the `cpu` field, the task will not share CPU

Reviewer: ```suggestion
exclusively for the task. Unlike the `cpu` field, the task does not share CPU
```

time with any other tasks managed by Nomad on the client.

```hcl
resources {
  cores = 2
}
```

If `cores` and `cpu` are both defined in the same resource block, validation of

> the job will fail.

Reviewer: ```suggestion
the job fails.
```

Refer to [How Nomad Uses CPU][concepts-cpu] for more details on Nomad's

> reservation of CPU resources.

Reviewer 2: Great idea linking it!
